docs/cloud-native-security/cspm-get-started-gcp.asciidoc

```diff
@@ -0,0 +1,82 @@
+[[cspm-get-started-gcp]]
+= Get started with CSPM for GCP
+
+[discrete]
+[[cspm-overview-gcp]]
+== Overview
+
+This page explains how to get started monitoring the security posture of your cloud assets using the Cloud Security Posture Management (CSPM) feature.
+
+.Requirements
+[sidebar]
+--
+* The CSPM integration is available to all {ecloud} users. On-premise deployments require an https://www.elastic.co/pricing[Enterprise subscription].
+* To view posture data, you need `read` privileges for the following {es} indices:
+** `logs-cloud_security_posture.findings_latest-*`
+** `logs-cloud_security_posture.scores-*`
+** `Logs-cloud_security_posture.findings`
+* The user who gives the CSPM integration GCP permissions must be a GCP project `admin`.
+--
+
+[discrete]
+[[cspm-setup-gcp]]
+== Set up CSPM for GCP
+
+To set up CSPM for GCP, first add the CSPM integration, then enable cloud account access.
+
+
+[discrete]
+[[cspm-add-and-name-integration-gcp]]
+=== Add your CSPM integration
+. From the Elastic Security *Get started* page, click *Add integrations*.
+. Search for `CSPM`, then click on the result.
+. Click *Add Cloud Security Posture Management (CSPM)*.
+. Give your integration a name that matches the purpose or team of the GCP account you want to monitor, for example, `dev-aws-account`.
+
+[discrete]
+[[cspm-set-up-cloud-access-section-gcp]]
+=== Set up cloud account access
+The CSPM integration requires admin access to your GCP project to
+
+For most users, the simplest option is to use a Google Cloud Shell script to automatically provision the necessary resources and permissions in your GCP account. This method, as well as two manual options, are described below.
+
+[discrete]
+[[cspm-set-up-cloudshell]]
+=== Cloud Shell script setup (recommended)
+
+. Under **Setup Access** select **Google Cloud Shell**.
+. Under **Where to add this integration**,
+  . If you want to monitor a GCP project where you have not yet deployed {agent}:
+    . Select **New Hosts**
+    . Name the {agent} policy. Use a name that matches the purpose or team of the cloud account or accounts you want to monitor. For example, `dev-gcp-account`.
+    . Click **Save and continue**, then **Add {agent} to your hosts**. The **Add agent** wizard appears and provides {agent} binaries, which you can download and deploy to a VM in your GCP account.
+  . If you want to monitor a GCP project where you have already deployed {agent}:
+    . Select **Existing hosts**.
+    . Select an agent policy that applies the GCP project you want to monitor.
+. Click **Save and continue**.
+. Copy the command that appears, then click **Launch Google Cloud Shell**. It opens in a new window.
+. Check the box to trust Elastic's `cloudbeat` repo, then click **Continue**
++
```

Comment:
Another clarification that I think we should add.

During the CloudShell provisioning, we add roles to [Google's default service account](https://cloud.google.com/compute/docs/access/service-accounts#google_apis_service_agent), which is utilized by the deployment manager for custom role creation and attaching SA to a compute instance.

However, after the deployment is finalized, we take the step of removing these roles from the service account. An issue that might arise is when a user attempts to delete the deployment but encounters a problem because the deployment manager lacks the necessary permissions.

In such instances, users should consider adding the specified predefined roles to the service account.
- [Project IAM Admin](https://cloud.google.com/iam/docs/understanding-roles#resourcemanager.projectIamAdmin)
- [Role Administrator ](https://cloud.google.com/iam/docs/understanding-roles#iam.roleAdmin)

Reply:
> During the CloudShell provisioning, we add roles to [Google's default service account](https://cloud.google.com/compute/docs/access/service-accounts#google_apis_service_agent), which is utilized by the deployment manager for custom role creation and attaching SA to a compute instance. However, after the deployment is finalized, we take the step of removing these roles from the service account.

@benironside can we include the text above as a note in this automated section?